csrc/fusion_segmenter.cpp

```diff
@@ -3603,6 +3617,10 @@ class MergeUpAndDownCast {
       SegmentedGroup* group = to_visit.front();
       to_visit.pop_front();
 
+      if (group->exprs().empty()) {
```

Comment:
Not strictly related, but while testing this PR, I was hitting a bug that should been easily spotted if this was in place.